server/src/main/java/org/apache/druid/client/coordinator/CoordinatorClientImpl.java

```diff
@@ -156,6 +163,25 @@ public ListenableFuture<List<DataSourceInformation>> fetchDataSourceInformation(
     );
   }
 
+  @Override
+  public ListenableFuture<CloseableIterator<DataSegment>> fetchBootstrapSegments()
+  {
+    final String path = "/druid/coordinator/v1/metadata/bootstrapSegments";
+    return FutureUtils.transform(
+        client.asyncRequest(
+            new RequestBuilder(HttpMethod.POST, path),
+            new InputStreamResponseHandler()
+        ),
+        in -> new JsonParserIterator<>(
+            // We specifically use LoadableDataSegment instead of DataSegment so the callers can correctly load the
```

Comment:
Since a `LoadableDataSegment` is a `DataSegment` and this iterator is materialized on the client side, we don't need to be explicit with the sub-type other than this one place.

Reply:
Clarified the comment a wee bit. Also, added a test `test_fetchBootstrapSegmentsAreLoadableWhenPruneLoadSpecIsEnabled()`.